### PR TITLE
Add HTTP transport

### PR DESCRIPTION
**Pull request by hasLeland**
Here's the fully merged in HTTP transport. I will merge if people say it's good.

### CODE REVIEW

File: gengokit/httptransport/httptransport.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+type Helper struct {
+	Methods           []*Method
+	PathParamsBuilder string
+}
+
+// NewHelper builds a helper struct from a service declaration. The other
+// "New*" functions in this file are there to make this function smaller and
+// more testable.
+func NewHelper(svc *doctree.ProtoService) *Helper {
+	pp := FormatCode(HTTPAssistFuncs)
```
Comment (hasAdamr) [inline gengokit/httptransport/httptransport.go]: I still thing mentioning where HTTPAssistFuncs comes from would be helpful.

File: gengokit/httptransport/httptransport_test.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			binding,
+		},
+	}
+	binding.Parent = meth
+
+	newMeth := NewMethod(&dmeth)
+	if got, want := newMeth, meth; !reflect.DeepEqual(got, want) {
+		t.Errorf("methods differ;\ngot  = %+v\nwant = %+v\n", got, want)
+	}
+	//t.Log(spew.Sdump(meth))
```
Comment (hasAdamr) [inline gengokit/httptransport/httptransport_test.go]: Little thing, commented out line

**Comment by hasAdamr** [comment]
LGTM, besides the little things but it would be fine without them.

**Comment by hasLeland** [comment]
Made the changes you recommended, also I went through it with golint and added comments till it had nothing else to say about the `httptransport/` folder. If you give the ok, I'll merge.

**Comment by hasAdamr** [comment]
LGTM

**Pull request merged**